components/brave_ads/core/internal/common/url/url_util_unittest.cc

```diff
@@ -14,133 +14,195 @@
   // Arrange
 
   // Act
 
   // Assert
-  EXPECT_TRUE(SchemeIsSupported(GURL("brave://rewards")));
+  EXPECT_TRUE(DoesSupportUrl(GURL("brave://rewards")));
 }
 
-TEST(BraveAdsUrlUtilTest, BraveSchemeWithRewardsHostNameAndPathIsSupported) {
+TEST(BraveAdsUrlUtilTest, DoesSupportBraveSchemeWithRewardsHostNameAndPath) {
   // Arrange
 
   // Act
 
   // Assert
-  EXPECT_TRUE(SchemeIsSupported(GURL("brave://rewards/foo")));
+  EXPECT_TRUE(DoesSupportUrl(GURL("brave://rewards/foo")));
+}
+
+TEST(BraveAdsUrlUtilTest, DoesNotSupportBraveSchemeWithSettingsHostName) {
+  // Arrange
+
+  // Act
+
+  // Assert
+  EXPECT_FALSE(DoesSupportUrl(GURL("brave://settings")));
 }
 
 TEST(BraveAdsUrlUtilTest,
-     BraveSchemeWithSettingsHostNameAndSearchEnginesPathIsSupported) {
+     DoesNotSupportBraveSchemeWithSettingsHostNameAndFooBarPath) {
   // Arrange
 
   // Act
 
   // Assert
-  EXPECT_TRUE(SchemeIsSupported(GURL("brave://settings/searchEngines")));
+  EXPECT_FALSE(DoesSupportUrl(GURL("brave://settings/foobar")));
 }
 
 TEST(BraveAdsUrlUtilTest,
-     BraveSchemeWithSettingsHostNameAndSearchPathIsSupported) {
+     DoesSupportBraveSchemeWithSettingsHostNameAndSearchEnginesPath) {
   // Arrange
 
   // Act
 
   // Assert
-  EXPECT_TRUE(SchemeIsSupported(GURL("brave://settings/search")));
+  EXPECT_TRUE(DoesSupportUrl(GURL("brave://settings/searchEngines")));
+}
+
+TEST(
+    BraveAdsUrlUtilTest,
+    DoesSupportBraveSchemeWithSettingsHostNameSearchEnginesPathAndSearchQuery) {
+  // Arrange
+
+  // Act
+
+  // Assert
+  EXPECT_TRUE(
+      DoesSupportUrl(GURL("brave://settings/searchEngines?search=foobar")));
+}
+
+TEST(
+    BraveAdsUrlUtilTest,
+    DoesNotSupportBraveSchemeWithSettingsHostNameSearchEnginesPathAndInvalidQuery) {
+  // Arrange
+
+  // Act
+  // Assert
+  EXPECT_FALSE(DoesSupportUrl(GURL("brave://settings/searchEngines?search")));
 }
 
 TEST(BraveAdsUrlUtilTest,
-     BraveSchemeWithSettingsHostNameAndFooBarPathIsNotSupported) {
+     DoesSupportBraveSchemeWithSettingsHostNameAndSearchPath) {
   // Arrange
 
   // Act
 
   // Assert
-  EXPECT_FALSE(SchemeIsSupported(GURL("brave://settings/foobar")));
+  EXPECT_TRUE(DoesSupportUrl(GURL("brave://settings/search")));
 }
 
-TEST(BraveAdsUrlUtilTest, BraveSchemeWithSettingsHostNameIsNotSupported) {
+TEST(BraveAdsUrlUtilTest,
+     DoesSupportBraveSchemeWithSettingsHostNameSearchPathAndSearchQuery) {
+  // Arrange
+
+  // Act
+
+  // Assert
+  EXPECT_TRUE(DoesSupportUrl(GURL("brave://settings/search?search=foobar")));
+}
+
+TEST(BraveAdsUrlUtilTest,
+     DoesNotSupportBraveSchemeWithSettingsHostNameSearchPathAndInvalidQuery) {
+  // Arrange
+
+  // Act
+
+  // Assert
+  EXPECT_FALSE(DoesSupportUrl(GURL("brave://settings/search?search")));
+}
+
+TEST(BraveAdsUrlUtilTest,
+     DoesNotSupportBraveSchemeWithSettingsHostNameAndQuery) {
+  // Arrange
+
+  // Act
+
+  // Assert
+  EXPECT_FALSE(DoesSupportUrl(GURL("brave://settings/?search=foobar")));
```

Comment:
Could you please add this test case too:
```
  EXPECT_FALSE(DoesSupportUrl(GURL("brave://settings/?search=foobar&another=value")));
```

Reply:
Great catch, thanks

Reply:
Fixed